README.md

```diff
@@ -21,30 +37,119 @@ from fastapi_nextauth_jwt import NextAuthJWT
 app = FastAPI()
 
 JWT = NextAuthJWT(
-    secret="y0uR_SuP3r_s3cr37_$3cr3t",
+    secret="y0uR_SuP3r_s3cr37_$3cr3t",  # Better to use environment variable
 )
 
 @app.get("/")
-async def return_jwt(jwt: Annotated[dict, Depends(JWT)]):
-    return jwt
+async def return_jwt_get(jwt: Annotated[dict, Depends(JWT)]):
+    return {"message": f"Hi {jwt['name']}. Greetings from fastapi!"}
+
+# For CSRF protection testing
+@app.post("/")
+async def return_jwt_post(jwt: Annotated[dict, Depends(JWT)]):
+    return {"message": f"Hi {jwt['name']}. Greetings from fastapi!"}
+```
+
+## Configuration Options
+
+### Essential Settings
+
+- **secret** (str): The secret key used for JWT operations. Should match `NEXTAUTH_SECRET` in your Next.js app.
+  ```python
+  JWT = NextAuthJWT(secret=os.getenv("NEXTAUTH_SECRET"))
+  ```
+
+### Security Options
+
+- **csrf_prevention_enabled** (bool): Toggle CSRF protection
+  - Defaults to `True` in development (`ENV=dev`), `False` otherwise
+
+- **csrf_methods** (Set[str]): HTTP methods requiring CSRF protection
+  - Default: `{'POST', 'PUT', 'PATCH', 'DELETE'}`
+
+### Cookie Configuration
+
+- **secure_cookie** (bool): Enable secure cookie attributes
+  - Default: `True` (when `NEXTAUTH_URL` starts with https)
+
+- **cookie_name** (str): Session token cookie name
+  - Default: `"__Secure-next-auth.session-token"` (when secure_cookie is True)
+  - Default: `"next-auth.session-token"` (when secure_cookie is False)
+  - For Auth.js (latest), use: `"__Secure-authjs.session-token"` (when secure_cookie is True)
+
+- **csrf_cookie_name** (str): CSRF token cookie name
+  - Default: `"__Host-next-auth.csrf-token"` (when secure_cookie is True)
+  - Default: `"next-auth.csrf-token"` (when secure_cookie is False)
+  - For Auth.js (latest), use: `"__Host-authjs.csrf-token"` (when secure_cookie is True)
+
+> [!TIP]
+> If you're using the latest version of Auth.js, here's the recommended configuration:
+> ```python
+> JWT = NextAuthJWT(
+>     secret=os.environ["AUTHJS_SECRET"],
+>     cookie_name="__Secure-authjs.session-token",
+>     csrf_cookie_name="__Host-authjs.csrf-token",
+> )
+> ```
```

Comment:
The code still needs to be adapted to use AUTHJS_SECRET by default, but the cookie names should already be correct by default